controllers/controller.js

```diff
@@ -10,42 +10,33 @@ mongoose.Promise = require('bluebird');
 mongoose.connect('mongodb://test:test@ds257627.mlab.com:57627/first-blog-mern', { useMongoClient: true });
 
 const dataService = new service();
-module.exports = app => {
-
-//    const homeFunc = (req, res) => {
-//        Post.find({}, function(err, data){
-//            if (err) throw (err);
-//            res.render('posts', {posts: data});
-//        });
-//    };
+module.exports = (app) => {
     
-    const homeFunc = (req, res) => {
-        const data = dataService.getAllPosts();
-        console.log(data);
-        res.render('posts', {posts: data});
-    };
 
-    app.get('/', homeFunc);
 
+    app.get(['/', '/posts'], (req, res) => {
+        dataService.getAllPosts().then(posts => {
+            res.render('posts', {posts: posts});
+        }).catch(
+            res.status(500);
+        );
+    });
     
-    app.get('/posts', homeFunc);
-
-    app.get('/posts/:post_id', function(req, res){
-        Post.find({_id: req.params.post_id}, function(err, data){
-            if (err) throw (err);
-            res.render('post_detail', {post: data});
+    app.get('/posts/:post_id', (req, res) => {
+        dataService.getOnePost(req.params.post_id).then((post) => {
+            res.render('post_detail', {post: post});
         });
     });
 
-    app.get('/post_new', function(req, res){
+    app.get('/post_new', (req, res) => {
         res.render('post_new');
     });
 
-    app.post('/post_new', urlencodedParser, function(req, res){
+    app.post('/post_new', urlencodedParser, (req, res) => {
         // get data from view and add to mongodb
-        const newPost = Post(req.body).save(function(err, data){
-            if (err) throw (err);
+        dataService.postNewPost(req.body).then((post) => {
```

Comment:
`req.body` is not sanitized. A user should not be able to specify the post's `id` field